src/core/lightning/index.ts

```diff
@@ -829,12 +827,17 @@ export const LightningMixin = (superclass) =>
     }
 
     async updatePendingInvoices(lock) {
-      // TODO
-      // const currency = "BTC"
-
-      const invoices = await InvoiceUser.find({ uid: this.user._id, paid: false })
+      const pendingInvoicesRes = this.invoices.findPendingByWalletId(this.user._id)
+      if (isRepoError(pendingInvoicesRes)) {
+        this.logger.error(
+          { walletId: this.user._id, error: pendingInvoicesRes },
+          "finish updating pending invoices with error",
+        )
+        return
+      }
 
-      for (const { _id: hash, pubkey: pubkeyCached } of invoices) {
+      const invoices = pendingInvoicesRes as AsyncGenerator<WalletInvoice>
```

Comment:
if this cast is needed there is a problem somewhere.
I think if you use :
```
if (pendingInvoicesRes instanceof Error) {
}
```

instead of `isRepoError` you won't need these casts. We definitely shouldn't use casts this way because it will mask bugs that would be caught statically.

Reply:
cast is only for eslint, we cant just repeat code because eslint for TS does not work properly.

Reply:
the other solution is worst, declare `pendingInvoicesRes` and then assign (that fix the eslint problem and we dont need cast) do you prefer that?

Reply:
I don't understand... lets look at code review